Extract oauth_manager from app.js

autra opened the pull request:
This commit introduces the OauthManager, which manage the oauth process and exposes a promise-based API to start it and get results.

@fabi1cazenave this commits attempts at making things clearer and have a better separation-of-concern in this app. Please R?

Commit:
Extract oauth_manager from app.js

This commit introduces the OauthManager, which manage the oauth process and
exposes a promise-based API to start it and get results.

fabi1cazenave commented on js/app.js:
```diff
@@ -122,52 +63,57 @@ window.addEventListener('DOMContentLoaded', function() {
+      }
+    }
+    var message =
+    showMessage(
```
```javascript
var message = showMessage(
```

fabi1cazenave commented on js/app.js:
```diff
@@ -122,52 +63,57 @@ window.addEventListener('DOMContentLoaded', function() {
+        <li>${nbUpdated} contact(s) updated</li>
+        <li>${nbDeleted} contact(s) deleted</li>
+      </ul>`
+    );
```
why don’t you put everything in a single string literal?

```javascript
    var message = showMessage(`
      <h2>Sync successfully finished:</h2>
      Summary:<br>
      <ul>
        <li>${nbAdded} contact(s) added</li>
        <li>${nbUpdated} contact(s) updated</li>
        <li>${nbDeleted} contact(s) deleted</li>
      </ul>
    `);
```
However, if that’s more than a debug message, we’ll need a proper l10n support — which will handle the pluralization properly.

fabi1cazenave replied:
(I know it was already like this before, but I didn’t get to review it back then ;-))

fabi1cazenave commented on js/app.js:
```diff
@@ -44,55 +19,21 @@ window.addEventListener('DOMContentLoaded', function() {
-    } else {
-      disableImport();
-    }
+    });
```
This looks like a rather surprising use of mozL10n. I need some time to figure out what you intend to do.

autra replied:
Maybe I can try to explain what was my intention: I want to change the wording of this button when we have a valid authorization token vs when we have not. I first tried to change only the data-l10n-id but I figured that mozL10n did not pick my change up at runtime (I expected that). So I thought that I needed to tell mozL10n that this element needs to be translated again.

My knowledge of l10n.js is limited, so there may very well be a better way. Alternatively, I can have 2 elements, one with the "Authorize", one with the "Reauthorize" and hide/show them when needed. WDYT?

fabi1cazenave replied:
Setting the data-l10n-id in the JS feels wrong, and it makes the initialization really painful. So yeah, I’d prefer two elements (or two spans within the same button + a class, whatever works).

fabi1cazenave commented on js/app.js:
```diff
@@ -6,36 +6,11 @@ window.addEventListener('DOMContentLoaded', function() {
-      `&client_id=${clientId}&approval_prompt=force&state=friends`;
-    oauthWindow = window.open(url, '', 'dialog');
+  authorizeButton.onclick = () => {
+    OauthManager.start().then(enableImport).catch(disableImport);
```
If you capitalize it, then let’s call it `OAuthManager` please — see http://oauth.net/

fabi1cazenave commented on js/oauth_manager.js:
```diff
@@ -0,0 +1,107 @@
+    tokenValidity = new Date(localStorage.getItem(TOKEN_VALIDITY_KEY));
+  }
+
+  function startOauth() {
```
`startOAuth` please, while you’re at it

autra replied:
yep, I've just run `sed -i 's;Oauth;OAuth;' $(git grep --name-only Oauth` on my local copy ;-)

Commit:
Correct capitalization of OAuth

Commit:
Improve l10n use

This commit internationalize part of the application that were not.
It also tries to use more correctly l10n.js.

autra commented on data/es.properties:
```diff
@@ -1,3 +1,3 @@
-app_title = Aplicación vacía privilegiada
+app-title = Aplicación vacía privilegiada
```
In gaia, they use `-` and no `_` as far as I can tell.

fabi1cazenave replied:
there’s no rule for that. You can use snake_case, camelCase, or hyphens if you really have no taste. :-D

autra replied:
Ok, so I will be lazy :-)

I've looked at sms, and they are using this form very consistently. You know who to complain to ;-)

autra commented:
@fabi1cazenave I've pushed 2 more commits. OAuth is now correctly capitalized.

For my use of l10n.js, I'm waiting for your input on how to use it correctly!

autra commented:
Misclicked....

fabi1cazenave commented on js/app.js:
```diff
@@ -44,55 +19,21 @@ window.addEventListener('DOMContentLoaded', function() {
-    enableImport();
+    navigator.mozL10n.once(() => {
+      navigator.mozL10n.translate(authorizeButton);
+    });
```
This is the part that really annoys me, and for which I’d much prefer a two-elements solution.

autra replied:
Ok, that's fine with me.

For my education, how do you indicate to l10n.js to translate a fragment you add dynamically to the dom?

autra commented:
@fabi1cazenave I have added one element for each wording of the (re)authorize button. Please r? again :-)

autra commented:
Let's timeout that and open issues for any followups.
